whedon.rb

```diff
@@ -108,12 +115,66 @@ def robawt_respond
   when /\A@whedon assignments/i
     reviewers, editors = assignments
     respond erb :assignments, :locals => { :reviewers => reviewers, :editors => editors, :all_editors => @config.editors }
+  when /\A@whedon generate pdf/i
+    puts "Attempting to compile PDF"
+    respond "```\nPDF at: #{process_pdf}\n```"
   end
 end
 
 # How Whedon talks
-def respond(comment)
-  settings.github.add_comment(@nwo, @issue_id, comment)
+def respond(comment, nwo=nil, issue_id=nil)
+  nwo ||= @nwo
+  issue_id ||= @issue_id
+  settings.github.add_comment(nwo, issue_id, comment)
+end
+
+# Download and compile the PDF
+def process_pdf
+  puts "In #process_pdf"
+  # TODO refactor this so we're not passing so many arguments to the method
+
+  pdf_path = WhedonWorker.new.perform(@config.papers, @config.site_host, @config.site_name, @nwo, @issue_id)
+
+  puts "Creating Git branch"
+  create_or_update_git_branch
+
+  puts "Uploading #{pdf_path}"
+  create_git_pdf(pdf_path)
+  # WhedonWorker.perform_async(@config.papers, @config.site_host, @config.site_name, @nwo, @issue_id)
+end
+
+# GitHub stuff (to be refactored!)
+
+def get_master_ref
+  settings.github.refs(@config.papers).select { |r| r[:ref] == "refs/heads/master" }.first.object.sha
+end
+
+# Create or update branch
+def create_or_update_git_branch
+  id = "%05d" % @issue_id
+
+  begin
+    # If the PDF is there already then delete it
+    settings.github.contents(@config.papers, :path => "10.21105.joss.#{id}.pdf", :ref => "heads/joss.#{id}")
+    blob_sha = settings.github.contents(@config.papers, :path => "10.21105.joss.#{id}.pdf", :ref => "heads/joss.#{id}").sha
+    settings.github.delete_contents(@config.papers,
```

Comment:
We should probably make sure we don't accidentally delete files that have been merged into `master` (i.e. if someone comes back to the repository and rebuilds the old paper)